tools/add_triage_signature.py

```diff
@@ -383,8 +384,87 @@ class StorageDetailSignature(Signature):
 
-    def _update_ticket(self, url, issue_key, should_update=False):
+    class SmartctlExitCode(Flag):
+        """
+        Taken from `man smartctl` "EXIT STATUS" section
+        """
+        COMMANDLINE_DID_NOT_PARSE = auto()
+        DEVICE_OPEN_FAILED = auto()
+        SOME_ATA_COMMAND_FAILED_OR_SMART_CHECKSUM_ERROR = auto()
+        DISK_FAILING = auto()
+        PREFAIL_ATTRIBUTRES_UNDER_THRESHOLD = auto()
+        DISK_OK_SOME_ATTRIBUTES_UNDER_THRESHOLD = auto()
+        LOG_HAS_RECORDS_OF_ERRORS = auto()
+        SELF_TEST_LOG_HAS_RECORDS_OF_ERRORS = auto()
+
+    @classmethod
+    def _parse_smart_internal(cls, smart):
+        output = []
+        smartctl = smart["smartctl"]
+        exit_code = smartctl["exit_status"]
+
+        if exit_code != 0:
+            output.append("non-zero smartctl exit code: " + ", ".join(flag.name for flag in cls.SmartctlExitCode
+                                                                      if flag in cls.SmartctlExitCode(exit_code)))
+            if cls.SmartctlExitCode(exit_code) == cls.SmartctlExitCode.SOME_ATA_COMMAND_FAILED_OR_SMART_CHECKSUM_ERROR:
+                if "smart_status" not in smart:
+                    output.append(
+                        "{color:green}*this typically happens on virtual disks*{color} that don't actually have S.M.A.R.T. information,"
+                        " but may also happen due to other reasons")
+
+        if "messages" in smartctl:
+            for message in smartctl["messages"]:
+                severity = message.get("severity", "unknown")
+                message_string = message.get("string", "")
+                output.append(f'smartctl message with severity "{severity}": "{message_string}"')
+
+        if "smart_status" in smart:
+            passed = smart["smart_status"]["passed"]
+            if passed:
+                if "model_name" in smart:
+                    if smart["model_name"] == "QEMU HARDDISK":
+                        return "{color:green}*Virtual QEMU HARDDISK pretending to have S.M.A.R.T.*{color}"
+
+                output.append("{color:green}*Disk S.M.A.R.T. status OK*{color}")
+            else:
+                output.append("{color:red}*Disk S.M.A.R.T. status BAD*{color}")
+
+        if "power_cycle_count" in smart:
+            output.append(f"disk has been power cycled {smart['power_cycle_count']} times throughout its life")
+
+        if "power_on_time" in smart:
+            if "hours" in smart["power_on_time"]:
+                hours = smart["power_on_time"]["hours"]
+                output.append(
+                    f"disk has been powered on for {hours // 24} days and {hours % 24} hours throughout its life")
+
+        if "ata_smart_attributes" in smart:
+            table = smart["ata_smart_attributes"].get("table", [])
+            interesting = ("Program_Fail_Count", "Erase_Fail_Count", "Offline_Uncorrectable")
```

Comment:
maybe rename to ``relevant_attributes``

Reply:
Meet you halfway `interesting_attributes`